src/server/scripts/Northrend/Naxxramas/boss_razuvious.cpp

```diff
@@ -113,33 +134,112 @@ class boss_razuvious : public CreatureScript
                     case EVENT_SHOUT:
                         DoCastAOE(SPELL_DISRUPTING_SHOUT);
-                        events.ScheduleEvent(EVENT_SHOUT, 25000);
+                        events.ScheduleEvent(EVENT_SHOUT, 16 * IN_MILLISECONDS);
                         return;
                     case EVENT_KNIFE:
                         if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 45.0f))
                             DoCast(target, SPELL_JAGGED_KNIFE);
-                        events.ScheduleEvent(EVENT_KNIFE, 10000);
-                        return;
-                    case EVENT_COMMAND:
-                        DoPlaySoundToSet(me, SOUND_COMMND);
-                        events.ScheduleEvent(EVENT_COMMAND, 40000);
+                        events.ScheduleEvent(EVENT_KNIFE, urandms(10,15));
                         return;
                 }
             }
 
             DoMeleeAttackIfReady();
         }
+
+        void Reset() override
+        {
+            SetCombatMovement(false);
+            _Reset();
+        }
     };
 
 };
 
+class npc_dk_understudy : public CreatureScript
+{
+    public:
+        npc_dk_understudy() : CreatureScript("npc_dk_understudy") { }
+
+        struct npc_dk_understudyAI : public ScriptedAI
+        {
+            npc_dk_understudyAI(Creature* creature) : ScriptedAI(creature), _instance(creature->GetInstanceScript()), bloodStrikeTimer(0) { }
+
+            void EnterCombat(Unit* /*who*/) override
+            {
+                me->SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_ONESHOT_NONE);
+                if (Creature* razuvious = ObjectAccessor::GetCreature(*me, _instance->GetGuidData(DATA_RAZUVIOUS)))
+                    razuvious->AI()->DoZoneInCombat(nullptr, 250.0f);
+            }
+
+            void JustReachedHome() override
+            {
+                if (_instance->GetBossState(BOSS_RAZUVIOUS) == DONE)
+                    me->DespawnOrUnsummon();
+                else
+                    ScriptedAI::JustReachedHome();
+            }
+
+            void UpdateAI(uint32 diff) override
+            {
+                if (!me->isPossessedByPlayer() && !UpdateVictim())
+                    return;
+
+                if (!me->isPossessedByPlayer())
+                {
+                    if (diff < bloodStrikeTimer)
+                        bloodStrikeTimer -= diff;
+                    else
+                        DoCastVictim(SPELL_UNDERSTUDY_BLOOD_STRIKE);
+                }
+
+                DoMeleeAttackIfReady();
+            }
+
+            void OnCharmed(bool apply) override
+            {
+                ScriptedAI::OnCharmed(apply);
+                if (apply)
+                {
+                    if (!me->IsInCombat())
+                        EnterCombat(me->GetCharmer());
+                    me->StopMoving();
+                    me->SetReactState(REACT_PASSIVE);
+                    _charmer = me->GetCharmerGUID();
+                }
+                else
```

Comment:
It is accessed fine. If it weren't, the pets' react state would never react and they would stop fighting after the first mind control wears off. This does not happen.